src/storage/src/source/mysql/statistics.rs

```diff
@@ -81,70 +85,79 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
 
             tokio::pin!(resume_uppers);
 
-            let mut offset_known = None;
-            let mut offset_committed = None;
+            let prev_offset_known = Cell::new(None);
+            let prev_offset_committed = Cell::new(None);
+            let stats_output = RefCell::new(stats_output);
```

Comment:
Nothing out of the ordinary here either, just some Cell/RefCell dance so that the two processing loops can proceed independently. The logic is mainly unchanged